NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -800,21 +850,24 @@ def JetsFromLQ(T):
 	for n in range(T.nJet):
 		if  ((T.Jet_jetId[n] & 0x2) > 0) and T.Jet_pt[n]>30 and abs(T.Jet_eta[n])<2.4 : #morse only use jets that pass tightid, fixme check jet eta
 			m = TLorentzVector()
-			m.SetPtEtaPhiM(T.Jet_pt[n],T.Jet_eta[n],T.Jet_phi[n],0)
+			m.SetPtEtaPhiM(T.Jet_pt[n],T.Jet_eta[n],T.Jet_phi[n],T.Jet_mass[n])
 			jets.append(m)
-	for n in range(T.nGenPart):
-		pdg = T.GenPart_pdgId[n]
-		if pdg not in [4,-4]: #Get charm quarks
-			continue
-		motherIndex = T.GenPart_genPartIdxMother[n]
-		motherid = 0
-		if motherIndex>-1:
-			motherid = T.GenPart_genPartIdxMother[motherIndex]
-		if motherid not in [42,-42]:
-			continue
-		m = TLorentzVector()
-		m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],0.0)
-		genjets.append(m)
+
+	if isData==False:
+
+		for n in range(T.nGenPart):
+			pdg = T.GenPart_pdgId[n]
+			if pdg not in [4,-4]: #Get charm quarks
+				continue
+			motherIndex = T.GenPart_genPartIdxMother[n]
+			motherid = 0
+			if motherIndex>-1:
+				motherid = T.GenPart_genPartIdxMother[motherIndex]
+			if motherid not in [42,-42]:
+				continue
+			m = TLorentzVector()
+			m.SetPtEtaPhiM(T.GenPart_pt[n],T.GenPart_eta[n],T.GenPart_phi[n],T.GenPart_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.